### PR TITLE
ast: disallowing unknown metadata annotations for rego-v1

### PR DESCRIPTION
**Pull request by johanfylling**
New `allow_unknown_annotations` annotation introduced to opt-out.

Fixes: #5411

### CODE REVIEW

**Comment by johanfylling** [comment]
The `opa fmt --rego-v1` command needs to be updated to add the `allow_unknown_annotations` annotation to any METADATA block that contains unknown annotations. Whichever is merged last of this and the fmt [PR](https://github.com/open-policy-agent/opa/pull/6413) will need to add this.

File: ast/parser.go [hunk 1]
```diff
@@ -2198,27 +2198,29 @@ func (p *Parser) validateDefaultRuleArgs(rule *Rule) bool {
...
+	Scope                   string                 `yaml:"scope"`
+	Title                   string                 `yaml:"title"`
+	Entrypoint              bool                   `yaml:"entrypoint"`
+	Description             string                 `yaml:"description"`
+	Organizations           []string               `yaml:"organizations"`
+	RelatedResources        []interface{}          `yaml:"related_resources"`
+	Authors                 []interface{}          `yaml:"authors"`
+	Schemas                 []rawSchemaAnnotation  `yaml:"schemas"`
+	Custom                  map[string]interface{} `yaml:"custom"`
+	AllowUnknownAnnotations bool                   `yaml:"allow_unknown_annotations"`
```
[the remark this answers is not in the document]
Comment (ashutosh-narkar) [reply ast/parser.go:2210]: What about something simple like `additional_annotations`?

**Comment by ashutosh-narkar** [comment]
Closing for now as this is not planned for OPA v1.0.

**Comment by anderseknert** [comment]
In case anyone finds this and is looking to enforce this in their policies — [Regal](https://docs.styra.com/regal) allows this via the [invalid-metadata-attribute](https://docs.styra.com/regal/rules/bugs/invalid-metadata-attribute) rule.

**Pull request closed without merging**